python/interpret_community/mimic/mimic_explainer.py

```diff
@@ -304,14 +313,86 @@ def __init__(self, model, initialization_examples, explainable_model, explainabl
         if isinstance(training_data, DenseData):
             training_data = training_data.data
 
+        self._original_eval_examples = None
```

Comment:
this is quite a bit of logic to put inside mimic explainer, I'm really wondering how we could simplify this as mimic explainer is already quite complicated